layers/meta-balena-compulab/recipes-support/resin-init/resin-init-board/resin-init-board

```diff
@@ -23,5 +23,20 @@ then
   fi
 fi
 
+# Initialize the fan
+if [ -d /sys/class/pwm/pwmchip1 ]
+then
+  echo 0 > /sys/class/pwm/pwmchip1/export
+
+  sleep 1
```

Comment:
This seems arbitrary. This type of configuration is really better done in a udev rule, so that when the node is detected in the sysfs the configuration takes place. That probably is more robust.

If the PWM takes longer to appear than ~1s the fan will not be initialized and that may be a problem for the device. Also, if it appears quickly this is introducing 1s of boot delay that could be avoided by a dynamic udev rule.

Anyway, it can also be improved later on I guess.

Reply:
Thanks for the suggestion. I just copy/pasted the backlighting initialization script, but you are right. I'll learn how to do that properly.